src/app/Library/CrudPanel/Traits/Tabs.php

```diff
@@ -106,56 +106,99 @@ public function isLastTab($label)
     }
 
     /**
+     * @deprecated Do not use this method as it will be removed in future versions!
+     * Instead, use $this->getElementsWithoutATab($this->getCurrentFields())
+     *
      * @return \Illuminate\Support\Collection
      */
     public function getFieldsWithoutATab()
     {
-        $all_fields = $this->getCurrentFields();
+        return $this->getElementsWithoutATab($this->getCurrentFields());
+    }
 
-        $fields_without_a_tab = collect($all_fields)->filter(function ($value, $key) {
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    public function getElementsWithoutATab(array $elements)
+    {
+        return collect($elements)->filter(function ($value) {
             return ! isset($value['tab']);
         });
+    }
 
-        return $fields_without_a_tab;
+    /**
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabFields(string $label)
+    {
+        return $this->getTabItems($label, 'fields');
     }
 
     /**
-     * @param $label
      * @return array|\Illuminate\Support\Collection
      */
-    public function getTabFields($label)
+    public function getTabColumns(string $label)
```

Comment:
It still smells fishy to me that we are trying to "abstract" the tab items, but creating a `getTab{Columns}` function. 

Where we call `getTabColumns()`, can't we just call `getTabItems($tabLabel, 'columns')` ? 

Also please, let's rename `$label` to `$tabLabel`. 👍

Reply:
Haha I had the same smell. And I don't like the name either. 

But then I saw lower that `getTabFields()` would still exist, we have to keep it for backwards-compatibiliy. So I think Mauro just added `getTabColumns()` so that we have one for fields and one for columns. Equality.

Reply:
`getTabFields()` should be also deprecated in favor of using `getTabItems($tabLabel, 'fields')`. (yes, kept there to avoid BC with a notice that will be removed in future versions, the same as the other method that mentioned fields `getFieldsWithoutATab`). 

We still have the "connection" with fields and columns, but will not be in function names. One day we could use the tab system to also tab a group of widgets, or tab any arbitrary information if we abstract the "Tab Container" and re-use it. 

There is no sense in not deprecating that function while creating the `getTabItems()`. `Equality <> Garbage` 🙃 

If you think I am wrong let me know so that I can do a final test on the PR as it is, but I really think we shouldn't be adding this method. 

Cheers

Reply:
Ok - let’s deprecate getTabFields() and not add getTabColumns()